homeassistant/components/huawei_lte/__init__.py

```diff
@@ -387,6 +391,28 @@ def _update_router(*_: Any) -> None:
     for router_config in config.get(DOMAIN, []):
         domain_config[url_normalize(router_config.pop(CONF_URL))] = router_config
 
+    def service_handler(service) -> None:
+        """Apply a service."""
+        url = service.data.get(CONF_URL)
+        router = hass.data[DOMAIN].routers.get(url)
+        if not router:
+            _LOGGER.error("%s: router %s unavailable", service.service, url)
+            return
+
+        if service.service == SERVICE_CLEAR_TRAFFIC_STATISTICS:
+            result = router.client.monitoring.set_clear_traffic()
+            _LOGGER.debug("%s: %s", service.service, result)
+        elif service.service == SERVICE_REBOOT:
+            result = router.client.device.reboot()
+            _LOGGER.debug("%s: %s", service.service, result)
+        else:
+            _LOGGER.error("%s: unsupported service", service.service)
+
+    for service in (SERVICE_CLEAR_TRAFFIC_STATISTICS, SERVICE_REBOOT):
+        hass.services.async_register(
```

Comment:
Should these require admin access and so use async_register_admin_service from `helpers.service` ?

Reply:
Sounds good in principle, although I'm not that familiar with working with services in the first place: I'd like these services to be available for automations, for example "if there's no net access, reboot the router". But I guess admin services are available for automations, i.e. there's no call.context.user_id then (or it is an admin)?

Reply:
Automations are run as admin, yeah.